segmentstore/storage/impl/src/main/java/io/pravega/segmentstore/storage/impl/bookkeeper/BookKeeperLogFactory.java

```diff
@@ -169,5 +193,74 @@ private BookKeeper startBookKeeperClient() throws Exception {
                          .build();
     }
 
+    /**
+     * Recreate the Bookkeeper client if a given log exhibits MAX_CREATE_ATTEMPTS_PER_LOG creation attempts (as a proxy
+     * for Container recoveries) within the period of time defined in LOG_CREATION_INSPECTION_PERIOD.
+     *
+     * @param logId Id of the log being restarted.
+     */
+    private void tryResetBookkeeperClient(int logId) {
+        synchronized (this) {
+            if (logInitializationTracker.containsKey(logId)) {
+                // Account for a restart of the Bookkeeper log.
+                logInitializationTracker.get(logId).incrementLogCreations();
```

Comment:
Take `logInitializationTracker.get(logId)` and stash it in a local variable. Then compare it against null in the `if` above and reuse that variable. This should reduce the number of `get` calls into that HashMap.

Reply:
Done.